Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -576,33 +576,33 @@ lemma IsAlternating.sup_edge {u x : V} (halt : G.IsAlternating G') (hnadj : ¬G'
     · aesop
   · aesop
 
-lemma IsPerfectMatching.symmDiff_of_isAlternating (hM : M.IsPerfectMatching)
+lemma Subgraph.IsPerfectMatching.symmDiff_of_isAlternating (hM : M.IsPerfectMatching)
     (hG' : G'.IsAlternating M.spanningCoe) (hG'cyc : G'.IsCycles) :
     (⊤ : Subgraph (M.spanningCoe ∆ G')).IsPerfectMatching := by
   rw [Subgraph.isPerfectMatching_iff]
   intro v
-  simp only [toSubgraph_adj, symmDiff_def, sup_adj, sdiff_adj, Subgraph.spanningCoe_adj]
+  simp only [toSubgraph_adj, symmDiff_def, SimpleGraph.sup_adj, sdiff_adj, Subgraph.spanningCoe_adj]
```

Comment:
Probably `Subgraph.sup_adj` should be `protected`, right?

Reply:
I have pushed (on a separate branch) a first step to this. I thought doing `open Subgraph` explicitly would bring them into scope, but that doesn't seem to work.

I think I'm doing something wrong (or introducing this is not worth the trouble it causes in other areas): https://github.com/leanprover-community/mathlib4/commit/78af12b8aa4f718a6931a08ccd4f50fc2df1e2ad

Reply:
So, I think the way this works in practice is that `Subgraph.sup_adj` shadows `SimpleGraph.sup_adj`, and that's generally what we want, since if you're in the Subgraph namespace, you'll generally want the Subgraph one. 

The wip-commit doesn't compile yet, and there's a bunch more places where we'd need to add `Subgraph.` before the `sup_adj`.

As I see it now, I'd say the trade-off is fine as it currently is? (I'm good specifying the `SimpleGraph.` here to prevent doing it everywhere else)